### PR TITLE
WIP: report: configure features

### PR DESCRIPTION
**Pull request by connorjclark**
Configuration might be useful?

no impact on the [devtools](https://source.chromium.org/chromium/chromium/src/+/master:third_party/devtools-frontend/src/front_end/panels/lighthouse/LighthouseReportRenderer.ts;l=152?q=ReportUIFeatures%20%20f:devtools%20-f:out&ss=chromium) integration (it overrides some printing stuff which probably shouldn't be configurable)

the PSI integration would benefit slightly

### CODE REVIEW

File: lighthouse-core/report/html/renderer/report-ui-features.js [hunk 1]
```diff
@@ -88,43 +119,69 @@ class ReportUIFeatures {
...
-    // if dark is selected in the settings panel.
-    if (!this._dom.isDevTools() && window.matchMedia('(prefers-color-scheme: dark)').matches) {
-      turnOffTheLights = true;
+    if (this.featureSet.toggleDarkMode) {
+      const topbarLogo = this._dom.find('.lh-topbar__logo', this._document);
+      topbarLogo.addEventListener('click', () => this._toggleDarkTheme());
+
+      // Do not query the system preferences for DevTools - DevTools should only apply dark theme
+      // if dark is selected in the settings panel.
+      if (!this._dom.isDevTools() && window.matchMedia('(prefers-color-scheme: dark)').matches) {
```
Comment (paulirish) [inline lighthouse-core/report/html/renderer/report-ui-features.js:159]: is the idea that a devtools rendering config would have `toggleDarkMode: false` (instead of this isDevTools()) ?

File: lighthouse-core/report/html/renderer/report-ui-features.js [hunk 2]
```diff
@@ -88,43 +119,69 @@ class ReportUIFeatures {
...
-    this._setupMediaQueryListeners();
-    this._dropDown.setup(this.onDropDownMenuClick);
-    this._setupThirdPartyFilter();
-    this._setupElementScreenshotOverlay(this._dom.find('.lh-container', this._document));
-    this._setUpCollapseDetailsAfterPrinting();
-    this._resetUIState();
-    this._document.addEventListener('keyup', this.onKeyUp);
-    this._document.addEventListener('copy', this.onCopy);
+    if (this.featureSet.mediaQueryListeners) {
+      this._setupMediaQueryListeners();
```
Comment (paulirish) [inline lighthouse-core/report/html/renderer/report-ui-features.js:123]: driveby: this fn should be renamed to `applyNarrowClassSometimes` or something

**Review by paulirish — COMMENTED** [review]
if this file was ESM, then we could just export the `setupElemScreenshots` method? would that be equivalent?

File: lighthouse-core/report/html/renderer/report-ui-features.js (hunk 1 above)
Comment (connorjclark) [reply lighthouse-core/report/html/renderer/report-ui-features.js:159]: didnt consider that but ya, sure

**Comment by connorjclark** [comment]
> if this file was ESM, then we could just export the setupElemScreenshots method? would that be equivalent?

Ignoring the unsolved problem of us using ESM... I suppose? Although I prefer this method (shoving the entire interface into an options object) vs. making arbitrary methods "public".

**Comment by brendankenny** [comment]
```
/**
 * @typedef FeatureSet
 * @property {boolean} dropDownMenu
 * @property {boolean | {containerEl: Element}} elementScreenshotOverlay
 * @property {boolean} fireworks
 * @property {boolean} i18n
 * @property {boolean} mediaQueryListeners
 * @property {boolean} printing
 * @property {boolean} showMetricDescriptionsOnError
 * @property {boolean} stickyHeader
 * @property {boolean} thirdPartyFilter
 * @property {boolean} toggleDarkMode
 */
```

what of these features is PSI likely to end up using?

**Comment by connorjclark** [comment]
Shane agreed with the three features I listed above.

**Comment by connorjclark** [comment]
also: should move `this._saveGistCallback` from Viewer features to base features, so that DevTools can use it (if we figure out how to postMessage...)

**Comment by brendankenny** [comment]
the download, opening, then saving is unfortunate, but saving a gist requires having github sign in, so I'm not sure if that's ever going to work

**Comment by connorjclark** [comment]
closing for #12254
